# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerRegistrationParent.h
# Commit: 679bed36bc0b
# Full Hash: 679bed36bc0b4dbca32def451b6385e5a898368b
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2021-06-10 09:49:36
# Regressor Bug: 1564235
# File Overlap Count: 2
# Description:
#   Bug 1564235 - P3 Create IPC for NavigationPreload in PServiceWorkerRegistration.ipdl. r=dom-worker-reviewers,asuth
#   
#   This patch implements the IPC for NavigationPreload APIs.
#   The IPC methods are created on PServiceWorkerRegistration.
#   
# ==============================================================================

diff -r 789ce1a19e9f -r 679bed36bc0b dom/serviceworkers/ServiceWorkerRegistrationParent.h
--- a/dom/serviceworkers/ServiceWorkerRegistrationParent.h	Thu Jun 10 00:48:11 2021 +0000
+++ b/dom/serviceworkers/ServiceWorkerRegistrationParent.h	Thu Jun 10 00:48:11 2021 +0000
@@ -33,6 +33,17 @@
   mozilla::ipc::IPCResult RecvUpdate(const nsCString& aNewestWorkerScriptUrl,
                                      UpdateResolver&& aResolver) override;
 
+  mozilla::ipc::IPCResult RecvSetNavigationPreloadEnabled(
+      const bool& aEnabled,
+      SetNavigationPreloadEnabledResolver&& aResolver) override;
+
+  mozilla::ipc::IPCResult RecvSetNavigationPreloadHeader(
+      const nsCString& aHeader,
+      SetNavigationPreloadHeaderResolver&& aResolver) override;
+
+  mozilla::ipc::IPCResult RecvGetNavigationPreloadState(
+      GetNavigationPreloadStateResolver&& aResolver) override;
+
  public:
   NS_INLINE_DECL_REFCOUNTING(ServiceWorkerRegistrationParent, override);
 